libraries-3/src/test/java/com/baeldung/jnats/NatsClientLiveTest.java

```diff
@@ -1,115 +1,336 @@
+    public static Connection createConnectionWithReportNoResponders() throws IOException, InterruptedException {
+        return createConnectionWithReportNoResponders("nats://localhost:4222");
+    }
 
-        Message message = fooSubscription.nextMessage(200);
-        assertNotNull("No message!", message);
-        assertEquals("hello there", new String(message.getData()));
+    public static Connection createConnectionWithReportNoResponders(String uri) throws IOException, InterruptedException {
+        Options options = new Options.Builder().server(uri)
+            .connectionListener((connection, event) -> System.out.println("Connection Event: " + event))
+            .errorListener(new CustomErrorListener())
+            .reportNoResponders()
+            .build();
+        return Nats.connect(options);
+    }
 
-        client.publishMessage(message.getReplyTo(), message.getSubject(), "hello back");
+    static class CustomErrorListener implements ErrorListener {
+        @Override
+        public void errorOccurred(Connection conn, String error) {
+            System.err.println("Error Occurred: " + error);
+        }
 
-        message = barSubscription.nextMessage(200);
-        assertNotNull("No message!", message);
-        assertEquals("hello back", new String(message.getData()));
+        @Override
+        public void exceptionOccurred(Connection conn, Exception exp) {
+            System.err.println("Exception Occurred: " + exp);
+        }
     }
 
+//    public static void main(String[] args) throws Exception {
+//        Connection conn = createConnection("nats://localhost:4222");
+//        Thread.sleep(60000);
+//    }
 
-    private NatsClient connectClient() {
-        return new NatsClient();
+    @Test
+    public void givenNoSubscribers_whenPublishingMessages_thenItCannotBeDeterminedIfMessageIsReceived() {
+        try (Connection natsConnection = createConnection()) {
+            natsConnection.publish("mySubject", "myData".getBytes());
+        }
+        catch (Exception e) {
+            fail("There should not be an exception");
+        }
     }
 
     @Test
-    public void whenWildCardSubscription_andMatchTopic_MessageReceived() throws Exception {
+    public void givenNoSubscribersAndReportNoResponders_whenRequestReply_thenReceiveNoResponders() {
+        try (Connection natsConnection = createConnectionWithReportNoResponders()) {
+            natsConnection.publish("noOneIsSubscribed", null);
+            CompletableFuture<Message> f = natsConnection.request("noOneIsSubscribed", null);
+            f.get(WAIT_AT_MOST_MILLIS, TimeUnit.MILLISECONDS);
+            fail("There should be an exception");
+        }
+        catch (ExecutionException e) {
+            Throwable cause = e.getCause();
+            assertNotNull(cause);
+            assertInstanceOf(JetStreamStatusException.class, cause);
+            assertTrue(cause.getMessage().contains("503 No Responders Available For Request"));
```

Comment:
For complex assertions we use AssertJ, which has helpers like:
https://www.baeldung.com/assertj-exception-assertion

A test like this would usually be written like:
> assertThatThrownBy(() -> {
> }).hasCauseInstanceOf(JetStreamStatusException.class)
>  .hasMessageContaining("...");

Avoiding the need for the manual fail("There should be an exception");

(similar in a few other places throughout the test)